src/explorer/LeetCodeTreeDataProvider.ts

```diff
@@ -197,6 +230,18 @@ export class LeetCodeTreeDataProvider implements vscode.TreeDataProvider<LeetCod
         ].join(os.EOL);
     }
 
+    private updateTreeDataByProblem(problem: IProblem): void {
+        const origin: IProblem | undefined = this.allProblems.get(problem.id);
+        if (origin && origin.isFavorite !== problem.isFavorite) {
+            const problemIndex: number = this.treeData.Favorite.findIndex((p: LeetCodeNode) => Number(p.id) >= Number(problem.id));
```

Comment:
can we directly compare `p.id` with `problem.id`?

Reply:
The problems are originally in number-ascending order, thus the comparation in index searching should also be done this way. 
e.g. the favorite problems' ids are ["1", "4", "16", "23"], I want to delete the problem with id "23" and id is compared directly. Since "4" >= "23" === true, the search process stops at "4", thus the deletion is incorrect.

Reply:
IMO, the main problem here is that we store the Favorite problems as an array instead of a Map(id -> Problem).

So you may observe that we have to use `findIndex` and `splice` to add/remove items, which I think is hard to read.

Can we change it to a Map?

Reply:
Maybe it can be discussed in subsequent PRs, when updating `treeData[Category.Company/Tag]` field is necessary(delete problem from treeview after it is solved)?

Reply:
Yes, can be solved in another PR.